toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexService.kt

```diff
@@ -337,3 +312,20 @@ private object CrateExternalizer : DataExternalizer<CargoRegistryCrate> {
         return CargoRegistryCrate(versions)
     }
 }
+
+private data class ParsedVersion(
```

Comment:
Can we move it to CargoRegistryCrateVersion's companion as `ParsedVersion` has no use outside of the `crateFromJson` function?

Reply:
I suppose that it should be in this file, as it's specific to the format of the local index? `CargoRegistryCrateVersion` should be a general interface describing a crate version. For example, if we later port the crates.io resolver to this service interface, it should use `CargoRegistryCrateVersion`, but it shouldn't have to know about `ParsedVersion`.